proto/attestation/endorsement.proto

```diff
@@ -32,36 +32,64 @@ message TransparentReleaseEndorsement {
   bytes rekor_log_entry = 2;
 }
 
-message RootLayerEndorsements {
+message RootEndorsements {
   // The serialized TEE certificate(s). The details of the format and how the
   // certificate(s) are encoded into this byte array are implementation
   // specific. In case of AMD-SEV-SNP, as described in
   // https://www.amd.com/system/files/TechDocs/57230.pdf, there are three
   // different certificates packaged in two different files.
   bytes tee_certificate = 1;
 
-  TransparentReleaseEndorsement stage0_endorsement = 2;
-};
+  // Endorsement of the Stage0 binary.
+  TransparentReleaseEndorsement stage0 = 2;
+}
 
-message KernelLayerEndorsements {
-  oneof type {
-    TransparentReleaseEndorsement linux_kernel = 1;
-    TransparentReleaseEndorsement oak_restricted_kernel = 2;
-  }
+message KernelEndorsements {
+  TransparentReleaseEndorsement kernel_image = 1;
+  TransparentReleaseEndorsement kernel_setup_data = 2;
+  TransparentReleaseEndorsement init_ram_fs = 3;
+}
+
+message InitEndorsements {
+  TransparentReleaseEndorsement binary = 1;
+}
+
+message ApplicationEndorsements {
+  TransparentReleaseEndorsement binary = 1;
 }
 
-message CustomLayerEndorsements {
+message ContainerEndorsements {
   TransparentReleaseEndorsement binary = 1;
 }
 
+message OakRestrictedKernelEndorsements {
+  RootEndorsements root_layer = 1;
+  KernelEndorsements kernel_layer = 2;
+  ApplicationEndorsements application_layer = 3;
+}
+
+message OakContainersEndorsements {
+  RootEndorsements root_layer = 1;
```

Comment:
Here we could also have `Layer` in the proto name (as per my other comment)

Reply:
Done.